spring-integration-graphql/src/test/java/org/springframework/integration/graphql/outbound/GraphQlMessageHandlerTests.java

```diff
@@ -0,0 +1,481 @@
+
+		StepVerifier.create(this.updateRepository.current())
+				.expectNext(expected)
+				.expectComplete()
+				.verify();
+
+	}
+
+	@Test
+	@SuppressWarnings("unchecked")
+	void testHandleMessageForSubscriptionWithRequestInputProvided() {
+
+		String executionId = this.idGenerator.generateId().toString();
+		this.graphQlMessageHandler.setExecutionId(executionId);
+
+		StepVerifier verifier = StepVerifier.create(
+						Flux.from(this.resultChannel)
+								.map(Message::getPayload)
+								.cast(RequestOutput.class)
+								.mapNotNull(RequestOutput::getData)
+								.cast(SubscriptionPublisher.class)
+								.map(Flux::from)
+								.flatMap(data -> data)
+				)
+				.consumeNextWith(requestOutput -> {
+					Map<String, Object> results = (Map<String, Object>) requestOutput.getData();
+					assertThat(results).containsKey("results");
+
+					Map<String, Object> queryResult = (Map<String, Object>) results.get("results");
+					assertThat(queryResult)
+							.containsKey("id")
+							.containsValue("test-data-01");
+
+				})
+				.expectNextCount(9)
+				.thenCancel()
+				.verifyLater();
+
+		this.inputChannel.send(
+				MessageBuilder
+						.withPayload(new RequestInput("subscription { results { id } }", null, Collections.emptyMap(), null, executionId))
+						.build()
+		);
+
+		verifier.verify(Duration.ofSeconds(10));
+	}
+
+	@Test
+	void testHandleMessageForQueryWithInvalidPayload() {
+
+		this.inputChannel.send(
+				MessageBuilder
+						.withPayload(new Object())
+						.build()
+		);
+
+		Message<?> errorMessage = errorChannel.receive(10_000);
+		assertThat(errorMessage).isNotNull()
+				.isInstanceOf(ErrorMessage.class)
+				.extracting(Message::getPayload)
+				.isInstanceOf(MessageHandlingException.class)
+				.satisfies((ex) -> assertThat((Exception) ex)
+						.hasMessageContaining(
+								"'queryExpression' must not be null"));
+
+	}
+
+	@Test
+	void testHandleMessageForMutationWithInvalidPayload() {
+
+		this.inputChannel.send(
+				MessageBuilder
+						.withPayload(new Object())
+						.build()
+		);
+
+		Message<?> errorMessage = errorChannel.receive(10_000);
+		assertThat(errorMessage).isNotNull()
+				.isInstanceOf(ErrorMessage.class)
+				.extracting(Message::getPayload)
+				.isInstanceOf(MessageHandlingException.class)
+				.satisfies((ex) -> assertThat((Exception) ex)
+						.hasMessageContaining(
+								"'queryExpression' must not be null"));
+
+	}
+
+	@Test
+	void testHandleMessageForSubscriptionWithInvalidPayload() {
+
+		this.inputChannel.send(
+				MessageBuilder
+						.withPayload(new Object())
+						.build()
+		);
+
+		Message<?> errorMessage = errorChannel.receive(10_000);
+		assertThat(errorMessage).isNotNull()
+				.isInstanceOf(ErrorMessage.class)
+				.extracting(Message::getPayload)
+				.isInstanceOf(MessageHandlingException.class)
+				.satisfies((ex) -> assertThat((Exception) ex)
+						.hasMessageContaining(
+								"'queryExpression' must not be null"));
+
+	}
```

Comment:
I see 3 tests with the same content.
I don't think we need many of them since `operationExpression` logic is the same for any GraphQL operation setting on the message handler.